src/js/common/schemaform/save-load-actions.js

```diff
@@ -135,29 +139,33 @@ export function saveInProgressForm(formId, version, returnUrl, formData) {
       body
     }).then((res) => {
       if (res.ok) {
-        dispatch(setSaveFormStatus(SAVE_STATUSES.success));
-      } else {
+        // TODO update this when the backend api is updated to return the timestamp
+        // return res.json();
+        return Promise.resolve({ timestamp: moment().valueOf() });
+      }
+
+      return Promise.reject(res);
+    })
+    .then(({ timestamp }) => {
+      dispatch(setSaveFormStatus(SAVE_STATUSES.success, timestamp));
+    })
+    .catch((resOrError) => {
+      if (resOrError instanceof Response) {
         // TODO: If they've sat on the page long enough for their token to expire
         //  and try to save, tell them their session expired and they need to log
         //  back in and try again. Unfortunately, this means they'll lose all
         //  their information.
-        if (res.status === 401) {
+        if (resOrError.status === 401) {
           dispatch(setSaveFormStatus(SAVE_STATUSES.noAuth));
         } else {
           dispatch(setSaveFormStatus(SAVE_STATUSES.failure));
         }
-      }
-      return Promise.resolve(); // For unit testing
-    }).catch((error) => {
-      // Probably a network error has occurred
-      // TODO: Log this in GA or something
-      console.error('Error saving form:', error.message); // eslint-disable-line no-console
-      dispatch(setSaveFormStatus(SAVE_STATUSES.failure));
-      if (__BUILDTYPE__ === 'development') {
-        return Promise.reject(); // For unit testing
```

Comment:
I've updated all the `saveInProgressForm` tests to put expect statements inside `then` instead of `catch` and verified that the `then` callback with `expect` statements is run for each test. 

The way the code works (outside of unit tests) is that we are handling errors inside a `catch`, and continuing on normally, so I think that's what we should be testing against. We can test that the appropriate statuses were dispatched in the `catch` callback without having to use `catch` in the unit tests.

If we keep these rejections behind a flag, we could end up missing an error that triggers a rejected promise we weren't expecting, because the promise is artificially being rejected just for tests.